### DISCUSSION TITLE
[Errno 111] Connection refused when connecting to langfuse within docker compose

### DISCUSSION THREAD

**Discussion by bspetrov** [langfuse/langfuse#2597] [Support] [answered]
Hi Guys,

I am trying to implement a self-hosted version. I have added the 'langfuse-server- service from the docker-compose.yml file here in the repo and connected it to my existing db that I am using with my application.

Everything is okay. I can login inside the langfuse instance and I can even create projects, keys and etc however I cannot create any traces from my app using the SDK. I am getting "[Errno 111] Connection refused",

Tried some thins, even added the HOSTNAME=0.0.0., added all my docker services under the same network and still the same error. doing the auth_check() i amm receiving True. I am kind of unaware how to proceed further and what should I check..

Do you have any possible advise what can be checked more and where the problem might be coming from?

Here is my docker-compose.yml file:
```
version: '3.7'

networks:
  app-network:
    driver: bridge
    name: app-network

services:
  web:
    build:
      dockerfile:
        Dockerfile.local
    command: bash -c "uvicorn app.main:app --host 0.0.0.0 --port 5000 --reload"
    restart: on-failure
    networks:
      - app-network
    environment:
      DEBUG: 1
    ports:
      - 5000:5000
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - ./app:/code/app
    env_file:
      - .env
  postgres:
    image: ankane/pgvector:latest
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 10s
      timeout: 5s
      retries: 5
    hostname: postgres
    networks:
      - app-network
    ports:
      - "5432:5432"
    volumes:
      - ~/apps/postgres:/var/lib/postgresql@15/data:rw
    environment:
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_DB=llm_app
      - PGUSER=postgres
  langfuse-service:
    image: langfuse/langfuse:2
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - app-network
    ports:
      - "4000:3000"
    environment:
      - DATABASE_URL=postgresql://postgres:postgres@postgres:5432/langfuse
      - NEXTAUTH_SECRET=mysecret
      - SALT=mysalt
      - NEXTAUTH_URL=http://localhost:4000/
      - TELEMETRY_ENABLED=${TELEMETRY_ENABLED:-true}
      - LANGFUSE_ENABLE_EXPERIMENTAL_FEATURES=${LANGFUSE_ENABLE_EXPERIMENTAL_FEATURES:-false}
      - HOSTNAME=0.0.0.0
```

**Comment by marcklingen** [comment]
When auth_check is true, everything should work. Can you see database logs and application logs for this issue?

**Reply by bspetrov** [reply]
No errors related to any of that in any logs, _langfuse_ container is simply sitting in Starting.. Ready in 507ms nothing else..

**Reply by marcklingen** [reply]
What hostname did you configure in your sdks?

**Reply by bspetrov** [reply]
If you mean the HOSTNAME env var I tried without doing anything to that var - supposedly using the default (localhost) and also tried 0.0.0.0 - no result.

If you mean the hostname when I am calling the SDK -> _http://localhost:4000_  
also tried with 3000 as port still no result.

**Reply by marcklingen** [reply, marked as answer]
You can open Langfuse UI on http://localhost:3000?

The core problem here is probably that you want to access Langfuse from within the docker compose setup -- seems like a docker networking issue and unrelated to Langfuse itself.

I'd try the internal docker networking instead of localhost:300 as a hostname set in the SDK (in your web container), I think this should be `http://langfuse-service:3000` here

**Reply by bspetrov** [reply]
That worked just fine now, traces are being sent using the callback handler.

THANK YOU!

**Reply by marcklingen** [reply]
awesome, glad to hear this 🙌
